src/IonParserBinaryRaw.ts

```diff
@@ -213,28 +217,40 @@ export class ParserBinaryRaw {
         return LongInt.fromIntBytes(Array.prototype.slice.call(bytes), sign);
     }
 
-    private readUnsignedInt() : number {
-        var v = 0, b;
+    static readUnsignedIntFrom(input: BinarySpan, numberOfBytes: number) {
+        let value = 0, bytesRead = 0, byte;
+        if (numberOfBytes < 1)
+            return 0;
 
-        // they have to tell us the length
-        if (this._len < 1) return 0;
+        while (bytesRead < numberOfBytes) {
```

Comment:
I expanded lots of single-character variable names into words for readability. The logic is the same, however.